### DISCUSSION TITLE
Editing - Auswahlliste aus Database Domain - Order by greift nicht

### DISCUSSION THREAD

**Discussion by faulgurke** [e-netze/webgis-community#157] [Q&A]
Für's Editieren habe ich Auswahllisten für die Adresserfassung (Gemeinde, Straße, Hausnummer) konfiguriert (Editierbare Felder -> punktförmiges Thema ->Gemeindeauswahl -> optional: Database Domain).

Die Auswahlliste für die Straßen und die Hausnummern werden nicht sortiert, obwohl ein ORDER BY Feld definiert wurde.
Ich komme nicht drauf, was falsch ist - Hilfe!

(Anmerkung: Für die Auswahlliste Gemeinde und Auswahlliste Straße musste ich views anlegen, weil die Auswahllisten unter Verwendung der Adressen-Tabelle nicht funktioniert hat.

Konfiguration Auswahlliste Straße:
![database_domain_strasse](https://github.com/gis-eni/webgis-forum/assets/100088983/9e2500c9-ac73-4386-b6c8-5458a9f86c4a)

Konfiguration Auswahlliste Hausnummer:
![database_domain_hausnr](https://github.com/gis-eni/webgis-forum/assets/100088983/39cc7112-899d-40c9-b550-b6ee7e7ba884)

Ergebnis Straße:
![database_domain_ergebnis_strasse](https://github.com/gis-eni/webgis-forum/assets/100088983/4bd749ac-c81b-4b3f-b70d-fe377c6ebac4)

Ergebnis Hausnummer:
![database_domain_ergebnis](https://github.com/gis-eni/webgis-forum/assets/100088983/ce030b55-d325-4d33-be0b-39be304975f9)

**Comment by gis-eni** [comment]
Ich kann das Problem nicht nachvollziehen. Bei uns funktioniert die Sortierung der Edit Auswahllisten genau wie beschrieben. 
Wir schauen (bei unseren alten) Editprojekten auch auf Views (SQL Server). Diese Views lassen sich wie gewünscht sortieren.

Vielleicht liegt es an deinen Views? Schick einmal das Statement direkt gegen deine Datenbank:

``` sql
select strasse from view_temp_lfh_strassen where gemeinde='....' order by strasse
```

Passt da die Sortierung?

Und weil ich schreibe "in unseren alten Editierprojekten". Es ist eigentlich schlechte Praxis die Auswahllisten direkt aus der Datenbank zu holen. Gut Praxis ist es die Auswahllisten über dein WebService zB DataLinq zu holen. Ist ein bisserl mehr aufwand, erreicht aber leichter eine Trennung von WebApp zur Datenbank.

https://docs.webgiscloud.com/cloud/tutorials/cms/editing/fields_domains_datalinq.html

**Reply by gis-eni** [reply]
Und müsste die WHERE Bedingung nicht ``strasse='{{strassse}}'`` lauten?

**Reply by faulgurke** [reply]
Hab's auf DataLinq umgestellt ;-)

**Reply by gis-eni** [reply]
Und jetzt gehts?

**Reply by faulgurke** [reply]
Yess. Was mich allerdings verwirrt hat, ist dass die WHERE Bedingung strasse={{strasse}} heißen muss (ohne Hochkommata).
